fbw-common/src/wasm/systems/systems/src/structural_flex/wing_flex.rs

```diff
@@ -0,0 +1,2089 @@
+    gear_weight_on_wheels: LandingGearWeightOnWheelsEstimator,
+
+    total_lift: Force,
+
+    ground_weight_ratio: Ratio,
+}
+impl WingLift {
+    fn new(context: &mut InitContext) -> Self {
+        Self {
+            gear_weight_on_wheels: LandingGearWeightOnWheelsEstimator::new(context),
+            total_lift: Force::default(),
+            ground_weight_ratio: Ratio::default(),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext) {
+        let total_weight_on_wheels = self.gear_weight_on_wheels.total_weight_on_wheels();
+
+        let raw_accel_no_grav = context.vert_accel().get::<meter_per_second_squared>();
+        let cur_weight_kg = context.total_weight().get::<kilogram>();
+
+        let lift_delta_from_accel_n = raw_accel_no_grav * cur_weight_kg;
+        let lift_1g = 9.8 * cur_weight_kg;
+        let lift_wow = -9.8 * total_weight_on_wheels.get::<kilogram>();
+
+        let lift = if total_weight_on_wheels.get::<kilogram>() > 500. {
+            // Assuming no lift at low wind speed avoids glitches with ground when braking hard for a full stop
+            if context.true_airspeed().get::<knot>().abs() < 20. {
+                0.
+            } else {
+                (lift_1g + lift_wow).max(0.)
+            }
+        } else {
+            lift_1g + lift_delta_from_accel_n
+        };
+
+        self.total_lift = Force::new::<newton>(lift);
+
+        self.ground_weight_ratio = total_weight_on_wheels / context.total_weight();
+    }
+
+    fn total_plane_lift(&self) -> Force {
+        self.total_lift
+    }
+
+    // Outputs the fraction of the weight of the plane that is applied on ground.
+    //      0-> Plane not on ground  0.5-> half the weight of the plane on ground ...
+    fn ground_weight_ratio(&self) -> Ratio {
+        Ratio::new::<ratio>(self.ground_weight_ratio.get::<ratio>().max(0.).min(1.))
+    }
+}
+impl SimulationElement for WingLift {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.gear_weight_on_wheels.accept(visitor);
+
+        visitor.visit(self);
+    }
+}
+
+// Map that gives the mass of each wingflex node given the masses of the plane fuel tanks
+struct WingFuelNodeMapper<const FUEL_TANK_NUMBER: usize, const NODE_NUMBER: usize> {
+    fuel_tank_mapping: [usize; FUEL_TANK_NUMBER],
+}
+impl<const FUEL_TANK_NUMBER: usize, const NODE_NUMBER: usize>
+    WingFuelNodeMapper<FUEL_TANK_NUMBER, NODE_NUMBER>
+{
+    // For each fuel tank, gives the index of the wing node where fuel mass is added
+    fn new(fuel_tank_mapping: [usize; FUEL_TANK_NUMBER]) -> Self {
+        Self { fuel_tank_mapping }
+    }
+
+    fn fuel_masses(&self, fuel_tanks_masses: [Mass; FUEL_TANK_NUMBER]) -> [Mass; NODE_NUMBER] {
+        let mut masses = [Mass::default(); NODE_NUMBER];
+        for (idx, fuel) in fuel_tanks_masses.iter().enumerate() {
+            masses[self.fuel_tank_mapping[idx]] += *fuel;
+        }
+        masses
+    }
+}
+
+enum A380fuelTanks {
```

Comment:
Shouldn't the A380 specific enum and structs reside in the a380 systems crate?

Reply:
Done quickly by moving everything, but there's obviously one peace of abstraction still missing: I'd want to have a generic wing flex mechanism in common place, whereas now there's core logic in the A380 part.
Will do for now, will have time to think about this later!